client/src/components/AppCommandBar/AppCommandBar.test.js

```diff
@@ -0,0 +1,171 @@
+import { eventService } from "../../services/EventService";
+
+import initIcons from "../../services/IconService/IconService";
+
+initIcons();
+
+jest.mock("../../services/ConfigService");
+jest.mock("../../services/ApiService");
+jest.mock("../../services/EventService");
+
+const getAllTwins = jest.spyOn(apiService, "getAllTwins");
+const setConfig = jest.spyOn(configService, "setConfig");
+
+
+
+const mockSuccesResponse = { "Status": "Success"};
+
+const optionalComponentsState = [{
+    id: "console",
+    name: "Console",
+    show: false,
+    showProp: "showConsole"
+  },{
+    id: "output",
+    name: "Output",
+    show: false,
+    showProp: "showOutput"
+  }
+];
+
+let toggleCounter = 0;
+
+const toggleOptionalComponent = () => {
+  toggleCounter++;
+};
+
+let container = null;
+beforeEach(() => {
+  container = document.createElement("div");
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  PubSub.clearAllSubscriptions();
+  unmountComponentAtNode(container);
+  container.remove();
+  container = null;
+});
+
+
+test("delete all twins component", async () => {
+    // The <ModelViewerComponent /> component calls the config service and won't call the API unless the appAdtUrl is set
+    configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+    apiService.getAllTwins.mockResolvedValue(mockSuccesResponse);
+
+    act(() => {render(<AppCommandBar optionalComponents={optionalComponentsState}
+        optionalComponentsState={optionalComponentsState}/>, container);});
+    
+    const button = container.querySelector(".delete-button");
+    act(() => {
+      button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+    });
+
+    const deleteButton = await screen.findByTestId("deleteTwins");
+
+    act(() => {
+      deleteButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+    });
+
+    expect(getAllTwins).toHaveBeenCalledTimes(1);
+
+  });
+
+  test("switch environment", async () => {
+    // The <ModelViewerComponent /> component calls the config service and won't call the API unless the appAdtUrl is set
+    configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+    configService.setConfig.mockResolvedValue(mockSuccesResponse);
+
+    act(() => {render(<AppCommandBar optionalComponents={optionalComponentsState}
+        optionalComponentsState={optionalComponentsState}/>, container);});
+    
+    const button = container.querySelector(".sign-in");
+    act(() => {
+      button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+    });
+
+    let input = await screen.findByTestId("urlInput");
+    input.value = "https://test.api.scus.digitaltwins.azure.net";
+    Simulate.change(input);
+    expect(input.value).toBe("https://test.api.scus.digitaltwins.azure.net");
+    const saveButton = await screen.findByTestId("saveConfiguration");
+    act(() => {
+      saveButton.dispatchEvent(new MouseEvent("click", { bubbles: true }));
```

Comment:
I'm getting an error when running the test. It doesn't fail the test but would be good to fix.

```
    console.error
      Error: Not implemented: HTMLFormElement.prototype.submit
          at module.exports (/Users/jamescarpinter/depot/digital-twins-explorer/client/node_modules/jsdom/lib/jsdom/browser/not-implemented.js:9:17)
          at HTMLFormElementImpl.submit (/Users/jamescarpinter/depot/digital-twins-explorer/client/node_modules/jsdom/lib/jsdom/living/nodes/HTMLFormElement-impl.js:88:5)
          at HTMLFormElementImpl._doSubmit (/Users/jamescarpinter/depot/digital-twins-explorer/client/node_modules/jsdom/lib/jsdom/living/nodes/HTMLFormElement-impl.js:80:10)
          at HTMLButtonElementImpl._activationBehavior (/Users/jamescarpinter/depot/digital-twins-explorer/client/node_modules/jsdom/lib/jsdom/living/nodes/HTMLButtonElement-impl.js:23:14)
          at HTMLButtonElementImpl._dispatch (/Users/jamescarpinter/depot/digital-twins-explorer/client/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:240:26)
          at HTMLButtonElementImpl.dispatchEvent (/Users/jamescarpinter/depot/digital-twins-explorer/client/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:94:17)
          at HTMLButtonElement.dispatchEvent (/Users/jamescarpinter/depot/digital-twins-explorer/client/node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:231:34)
          at /Users/jamescarpinter/depot/digital-twins-explorer/client/src/components/AppCommandBar/AppCommandBar.test.js:103:18
          at batchedUpdates$1 (/Users/jamescarpinter/depot/digital-twins-explorer/client/node_modules/react-dom/cjs/react-dom.development.js:22380:12)
          at act (/Users/jamescarpinter/depot/digital-twins-explorer/client/node_modules/react-dom/cjs/react-dom-test-utils.development.js:1042:14) undefined
```